client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -133,6 +133,7 @@ public synchronized int getVersion(String subject, Schema schema)
       }
       int version = getVersionFromRegistry(subject, schema);
       schemaVersionMap.put(schema, version);
+      versionCache.put(subject, schemaVersionMap);
```

Comment:
We only need to do the put if schemaVersionMap was just created, right? So I think we could move this into the else block of the conditional at the top of this method.